server/src/omtLinkProvider.ts

```diff
@@ -0,0 +1,199 @@
+                            const newPath = resolve(dirname(uri), relPath);
+                            paths.push([key.substr(0, key.lastIndexOf('/*')), newPath]);
+                        }
+                    }
+                } catch (e) {
+                    // need to catch this error for when there is a JSON parsing error for example
+                    // we can still continue with other files
+                    console.error(e);
+                }
+                return paths;
+            }, []));
+    }
+
+    /**
+     * scan the document for imports and return DocumentLinks for
+     * absolute paths,
+     * relative paths,
+     * paths starting with a shorthand ('@shorthand/relativePath'),
+     * declared imports ('module: moduleName'). For declared imports the target will be undefined.
+     * The declared import links can be resolved using the data and the `resolveLink` function.
+     * @param document the OMT file we are scanning
+     * @param resolveShorthand function to resolve shorthand annotations at the start of import paths
+     */
+    private findOMTUrl(document: TextDocument, shorthands: Map<string, string>): DocumentLink[] {
+        const documentLinks: DocumentLink[] = [];
+        // so match after import: until we need any other (\w+): without any preceding spaces
+        let isScanning = false;
+        for (let l = 0; l <= document.lineCount - 1; l++) {
+            const line = getLine(document, l);
+            if (importMatch.exec(line)) {
+                isScanning = true; // start scannning lines for import paths after we enter an import block
+            } else if (isScanning) {
+                if (otherDeclareMatch.exec(line)) {
+                    break; // we can stop matching after the import block
+                } else {
+                    const uriMatch = omtUriMatch.exec(line);
+                    const diMatch = declaredImportMatch.exec(line);
+                    if (uriMatch) {
+                        // match[0] is the full match inluding the whitespace of match[1]
+                        // match[1] is the whitespace and optional quotes. both of which we don't want to include in the linked text
+                        // match[2] is the link text, including the @shorthands
+                        const link = replaceStart(uriMatch[2].trim(), shorthands);
+                        const url = isAbsolute(link) ? resolve(document.uri, link) : toAbsolutePath(document, link);
+                        documentLinks.push(
+                            createDocumentLink(l, uriMatch[1].length, uriMatch[2].length, url));
+                    }
+                    else if (diMatch) {
+                        // match[0] is the full line match including the trailing colon
+                        // match[1] is the whitespace prepending the import
+                        // match[2] is the text 'module:'
+                        // match[3] is the module name
```

Comment:
```suggestion
                        // match[2] is the module name
```
for the updated regex pattern

Reply:
tnx, changed in next commit